src/components/colorbar/attributes.js

```diff
@@ -143,18 +143,22 @@ module.exports = overrideAll({
             'In other cases the default is *hide past div*.'
         ].join(' ')
     }),
+
+    // ticklabelposition: not used directly, as values depend on orientation
+    // left/right options are for x axes, and top/bottom options are for y axes
     ticklabelposition: {
         valType: 'enumerated',
         values: [
             'outside', 'inside',
             'outside top', 'inside top',
+            'outside left', 'inside left',
+            'outside right', 'inside right',
             'outside bottom', 'inside bottom'
         ],
         dflt: 'outside',
-        description: [
-            'Determines where tick labels are drawn.'
-        ].join(' ')
+        description: 'Determines where tick labels are drawn.'
```

Comment:
Let's bring your comment above into the description, and clarify a little.
```suggestion
        description: [
            'Determines where tick labels are drawn relative to the ticks.',
            'Left and right options are used when `orientation` is *h*,',
            'top and bottom when `orientation` is *v*.'
        ].join(' ')
```